plugins/postgres/dbt/adapters/postgres/impl.py

```diff
@@ -112,10 +112,10 @@ def _link_cached_database_relations(self, schemas: Set[str]):
                 self.cache.add_link(referenced, dependent)
 
     def _get_catalog_schemas(self, manifest):
-        # postgres/redshift only allow one database (the main one)
+        # postgres/redshift(not ra3) only allow one database (the main one)
         schemas = super()._get_catalog_schemas(manifest)
         try:
-            return schemas.flatten()
+            return schemas.flatten(allow_duplicates=self.config.credentials.ra3_node)
```

Comment:
Postgres credentials don't have an `ra3_node` property, so this is raising an error and breaking tests. I think the right move here is to reimplement `_get_catalog_schemas` within [`redshift/impl.py`](https://github.com/fishtown-analytics/dbt/blob/develop/plugins/redshift/dbt/adapters/redshift/impl.py), including this RA3-specific logic.